app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java

```diff
@@ -358,6 +359,15 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
         }
     }
 
+    private void enableDeleteButton(boolean visibility) {
```

Comment:
Extracted the delete button states into this method to increase readability and reduce method complexity.